src/theory/quantifiers/sygus_inst.h

```diff
@@ -0,0 +1,76 @@
+/*********************                                                        */
+/*! \file sygus_inst.h
+ ** \verbatim
+ ** Top contributors (to current version):
+ **   Mathias Preiner
+ ** This file is part of the CVC4 project.
+ ** Copyright (c) 2009-2020 by the authors listed in the file AUTHORS
+ ** in the top-level source directory) and their institutional affiliations.
+ ** All rights reserved.  See the file COPYING in the top-level source
+ ** directory for licensing information.\endverbatim
+ **
+ ** \brief SyGuS instantiator class.
+ **/
+
+#include "cvc4_private.h"
+
+#ifndef CVC4__THEORY__QUANTIFIERS__SYGUS_INST_H
+#define CVC4__THEORY__QUANTIFIERS__SYGUS_INST_H
+
+#include <unordered_map>
+#include <unordered_set>
+
+#include "context/cdhashset.h"
+#include "theory/quantifiers/quant_util.h"
+
+namespace CVC4 {
+namespace theory {
+
+class QuantifiersEngine;
+
+namespace quantifiers {
+
+class SygusInst : public QuantifiersModule
```

Comment:
Add a few sentences describing the approach at a high level in a comment above the class here.